ironfish-rust/src/transaction/mod.rs

```diff
@@ -935,3 +947,44 @@ pub fn batch_verify_transactions<'a>(
         &SAPLING.mint_verifying_key,
     )
 }
+
+pub struct SecretShareConfig {
+    pub min_signers: u16,
+    pub max_signers: u16,
+    pub secret: Vec<u8>,
+}
+
+pub fn split_secret(
+    config: &SecretShareConfig,
+    identifiers: IdentifierList,
+    rng: &mut ThreadRng,
+) -> Result<(HashMap<Identifier, KeyPackage>, PublicKeyPackage), Error> {
+    let secret_key = SigningKey::deserialize(
+        config
+            .secret
+            .clone()
+            .try_into()
+            .map_err(|_| Error::MalformedSigningKey)?,
+    )?;
+
+    let (shares, pubkeys) = frost::keys::split(
+        &secret_key,
+        config.max_signers,
+        config.min_signers,
+        identifiers,
+        rng,
+    )?;
+
+    for (_k, v) in shares.clone() {
+        frost::keys::KeyPackage::try_from(v)?;
+    }
+
+    let mut key_packages: HashMap<_, _> = HashMap::new();
```

Comment:
should this `HashMap` use the specific `Identifier, KeyPackage` type? does it make a difference?

Reply:
Yea you can remove the `: HashMap<_, _>` and it will do type inference